ion/integration_test.go

```diff
@@ -46,6 +47,9 @@ func (i *ionItem) equal(o ionItem) bool {
 	if !cmpAnnotations(i.annotations, o.annotations) {
 		return false
 	}
+	if !cmpFieldNames(i.fieldName, o.fieldName) {
```

Comment:
This test requires that the two structs' field names be in the same order, but the Ion spec allows field names to be in any order when checking for equality.

Reply:
So probably I have to move this to ```cmpStruct``` function in ```cmp_test``` file. We know we have field name only when we are checking structs